[ch60779] [density chart] minor: Avoid blinking map when no actual difference is sent

tdesfont opened the pull request:
[ch60779](https://app.clubhouse.io/dataiku/story/60779/density-chart-chart-is-reloaded-when-showing-hiding-parameter-sections)
Avoid blinking map by comparing previous and next maptile ids, also setting maptile quality to highest quality

tdesfont committed:
[ch60779]
Avoid blinking map by comparing previous and next maptile ids, also setting maptile quality to highest quality

tdesfont committed:
geo-density-chart.js: Removed unused variable _tileMapProvider

tdesfont commented on resource/geo-density-chart.js:
```diff
@@ -84,23 +84,31 @@ function GeoDensityChart(){
         // iterate in the layers of the map
         _mapPointer.eachLayer(function(layer){
             if (layer.options.id){
```
Select the map layer using the key layer.options.id which is only available for map

Marsobad commented on resource/geo-density-chart.js:
```diff
@@ -84,23 +84,31 @@ function GeoDensityChart(){
+                console.log("[MapTile] Received previous maptile id:", layer.options.id);
                 let url;
-                _mapPointer.removeLayer(layer);
+                let idTileMap;
```
You could perhaps rename `idTileMap` to sth like `newMapBackground` because for me, this name makes it difficult to understand that it's actually the map background colour.

tdesfont replied:
I renamed `idTileMap` to `newBaseMap`, I think the light_all is more than just the color of the map it can be an id for an exotic map background

tdesfont replied:
Done in d2c7118

Marsobad commented on resource/geo-density-chart.js:
```diff
@@ -84,23 +84,31 @@ function GeoDensityChart(){
-                    tileSize: 512,
-                    zoomOffset: -1
-                }).addTo(_mapPointer).set;
+                if (layer.options.id !== idTileMap){
```
To make it easier to understand, I suggest to name the boolean `layer.options.id !== idTileMap`
Perhaps : `isBackgroundUpdated` or `hasBackgroundChanged`or `backgroundChange`

tdesfont replied:
Done in d2c7118, add precision on the two compared variable

Marsobad reviewed (changes requested):
- This fix solves the issue when you change parameters 
- I noticed that the view is re-scaled when you modify a **filter** or a **tooltip** while it's not the case for the scatter map. Is it intentional?

tdesfont committed:
[ch60779] Avoid map centering when not necessary
config-event.js: Creation of a config-event class variable to store necessity of centering the map
app.js: Using this newly created variable to condition the map centering

tdesfont committed:
[ch60779] Rename idTileMap to newBaseMap and set intermediate variable previousBaseMap to store value of layer.options.id

Marsobad reviewed (approved):
Looks great to me !
